desktop/core/src/desktop/js/apps/admin/ServerLogs/ServerLogsTab.tsx

```diff
@@ -0,0 +1,131 @@
+// Licensed to Cloudera, Inc. under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  Cloudera, Inc. licenses this file
+// to you under the Apache License, Version 2.0 (the
+// 'License'); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an 'AS IS' BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { useState, useEffect } from 'react';
+import { Spin, Alert } from 'antd';
+import ServerLogsHeader from './ServerLogsHeader';
+import './ServerLogsTab.scss';
+
+const ServerLogs: React.FC = (): JSX.Element => {
+  const [loading, setLoading] = useState(true);
+  const [logs, setLogs] = useState<string[]>([]);
+  const [noLogsFound, setNoLogsFound] = useState(false);
+  const [error, setError] = useState<string>('');
+  const [filter, setFilter] = useState<string>('');
+  const [hostName, setHostName] = useState<string>('');
+  const [wrapLogs, setWrapLogs] = useState(true);
+
+  useEffect(() => {
+    const fetchLogs = async () => {
+      try {
+        const response = await fetch('/api/v1/logs');
+        if (!response.ok) {
+          throw new Error(`HTTP error! Status: ${response.status}`);
+        }
+        const json = await response.json();
+        const fetchedLogs = json['logs'];
+        if (!fetchedLogs.length || fetchedLogs[0] === '') {
+          setNoLogsFound(true);
```

Comment:
is it required to have a state for `noLogsFound`? Can we calculate the length of the logs directly when it is required?